Updates for 2l2q channel

Anusreevijay769 opened the pull request.

YujiLee301 committed:
H4L updates

YujiLee301 committed:
“Fsr Recovery & Rochester correction”

YujiLee301 committed:
Jet Selection Updated

YujiLee301 committed:
Mela loaded

YujiLee301 committed:
Delete helloworld.h

YujiLee301 committed:
Delete NJettiness.h

YujiLee301 committed:
Delete NJettiness.cc

YujiLee301 committed:
Delete helloworld.cc

YujiLee301 committed:
Delete proxy.sh

YujiLee301 committed:
Delete condor_setup_lpc.py

YujiLee301 committed:
Delete JetSFMaker.py

YujiLee301 committed:
Delete HZZAnalysisModule.py

YujiLee301 committed:
Delete condor_setup_lxplus.py

YujiLee301 committed:
Add files via upload

YujiLee301 committed:
Remove redundancy

YujiLee301 committed:
Update README.md

YujiLee301 committed:
Add files via upload

YujiLee301 committed:
Update README.md

YujiLee301 committed:
Update README.md

YujiLee301 committed:
Update README.md

YujiLee301 committed:
RoccoR Added

YujiLee301 committed:
Path fixed

YujiLee301 committed:
Add files via upload

YujiLee301 committed:
Add files via upload

YujiLee301 committed:
Delete __init__.pyc

YujiLee301 committed:
Delete RoccoR.cc

YujiLee301 committed:
Delete RoccoR.h

YujiLee301 committed:
Update README.md

YujiLee301 committed:
Update H4LCppModule.py

ram1123 committed:
Update keep_and_drop.txt

ram1123 committed:
Update README.md

Following updates:

- Updated to `CMSSW_10_6_30` from `CMSSW_10_6_20`
- Updated to the latest version of official nanoAOD-tools
- Added back the instruction of batch job.

ram1123 committed:
Update README.md

YujiLee301 committed:
MELA updates

YujiLee301 committed:
Delete KalmanMuonCalibrationsProducer/interface directory

YujiLee301 committed:
Delete KalmanMuonCalibrationsProducer/plugins directory

YujiLee301 committed:
Delete KalmanMuonCalibrationsProducer/python directory

YujiLee301 committed:
Delete KalmanMuonCalibrationsProducer/src directory

YujiLee301 committed:
Delete BuildFile.xml

YujiLee301 committed:
Trigger selection added

YujiLee301 committed:
Trigger Updated

YujiLee301 committed:
Update keep_and_drop.txt with Mela branches

ram1123 committed:
Update keep_and_drop.txt

YujiLee301 committed:
Update keep_and_drop.txt

YujiLee301 committed:
Sync for Lepton

YujiLee301 committed:
Add Z selection

YujiLee301 committed:
Lepton Module Changed

YujiLee301 committed:
hard code removed

YujiLee301 committed:
Reading branch modified

ram1123 committed:
First update for 2l2q

ram1123 committed:
UPdate the printing style and fixed the flags

Anusreevijay769 committed:
Added additional variables for resolved category

ram1123 reviewed (commented):
- Add the resolved category information for ZZSystem
- Run over the sample `/GluGluHToZZTo2L2Q_M1000_TuneCP5_13TeV_powheg2_JHUGenV7011_pythia8/RunIISummer20UL18NanoAODv9-106X_upgrade2018_realistic_v16_L1v1-v1/NANOAODSIM`
   - Run with full statistics and get the cut-flow table


Cut flow table using miniAOD (From HIG-23-001)

| Cut string name                   | left nEvents | Change (%) |
|-----------------------------------|------------|---------|
| nEvents                           |  197000.00 | 167.18 |
| passed_trig                       |  187742.00 |  -4.70 |
| passed_n_2leps                    |  153719.00 | -18.12 |
| passed_flavor_charge              |  152717.00 |  -0.65 |
| passed_lepZpt_40_25               |  147278.00 |  -3.56 |
| passed_dR_lilj0point2             |  147278.00 |   0.00 |
| passed_Mll_4                      |  147275.00 |  -0.00 |
| passed_lepisolation               |  145438.00 |  -1.25 |
| passed_leptightID                 |  145083.00 |  -0.24 |
| passed_lepZmass40_180             |  139912.00 |  -3.56 |
| passed_lepZ                       |  139877.00 |  -0.03 |
| passed_lepclean                   |  138444.00 |  -1.02 |
| passed_jetpt30                    |  138444.00 |   0.00 |
| passed_nmergedjets_lepclean       |  136318.00 |  -1.54 |
| passed_mergedjetsPt200            |  136318.00 |   0.00 |
| passed_mergedjetEta2opint4        |  135786.00 |  -0.39 |
| passed_jeteta2opint4              |  127203.00 |  -6.32 |
| passed_dijetpt100                 |  125433.00 |  -1.39 |
| passed_mergedmass_40_180          |  120591.00 |  -3.86 |
| passed_MergedolonlyHiggs          |  109317.00 |  -9.35 |
| passed_ak4jetZm_40_180            |   99384.00 |  -9.09 |
| passed_particleNetZvsQCD0opint9   |   73733.00 | -25.81 |

ram1123 committed:
updated condor script

ram1123 commented on src/H4LTools.cc:
```diff
@@ -948,21 +948,26 @@ bool H4LTools::ZZSelection_2l2q(){
-            cut2l1J++;
-            cut2l1Jor2j++;
 
+            TLorentzVector Z2; //AV
```
Z2 is already defined in the .h file so you don't have to define this again.

ram1123 committed:
Added total number of events to print and in post-proc.py switched total entries to run to all

ram1123 committed:
Fixed the bug for FatJet selection; Updated the log summary

ram1123 committed:
src/H4LTools.ccMerge branch 'HZZ_Analysis_2l2q_v2_dev' of github.com:Anusreevijay769/nanoAOD_skim into HZZ_Analysis_2l2q_v2_devR

ram1123 committed:
Removed unnecessary lines

ram1123 committed:
Updated trigger list similar to HIG-23-001

Anusreevijay769 committed:
Merge branch 'HZZ_Analysis_2l2q_v2' into HZZ_Analysis_2l2q_v2_dev

ram1123 committed:
Merge branch 'HZZ_Analysis_2l2q_v2_dev' into HZZ_Analysis_2l2q_v2_devR

ram1123 commented on H4LCppModule.py:
```diff
@@ -88,6 +88,7 @@ def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
         self.out.branch("pTZ2",  "F")
         self.out.branch("etaZ2",  "F")
         self.out.branch("phiZ2",  "F")
+        self.out.branch("massZ2_2j",  "F")
```
You need to add all kinematic information for _2j case. 

Currently, its only mass. You need to add rapidity, energy, phi, etc.

Anusreevijay769 committed:
Merge pull request #1 from ram1123/HZZ_Analysis_2l2q_v2_devR

Fix for FatJet & Updated condor script

ram1123 commented on src/H4LTools.cc:
```diff
@@ -948,7 +951,15 @@ bool H4LTools::ZZSelection_2l2q(){
+            TLorentzVector Z2_2;//AV
+            Z2_1.SetPtEtaPhiM(Jet_pt[0], Jet_eta[0], Jet_phi[0], Jet_mass[0]);
+            Z2_2.SetPtEtaPhiM(Jet_pt[1], Jet_eta[1], Jet_phi[1], Jet_mass[1]);
+            Z2_2j = Z2_1 + Z2_2;
```
I think here, we will have some issue. 

Suppose in an event if there are less than two AK4 jets then this will give us error.

So, we need to protect this and skip the error if it appears.

ram1123 committed:
Update condor_setup_lxplus.py

To avoid confusion, I updated the name of the variables.

Also, I added all the variables we might update for each run in argparser.

ram1123 commented:
I tried to run this. Here is summary:

```
Total            :  197000 Events
PassTrig         :  183369 Events
Pass2eCut        :   59404 Events
Pass2muCut       :   79187 Events
Pass2lCut        :  138591 Events
Pass2l1JCut      :   96840 Events
Pass2l2jCut      :   23307 Events
Pass2l1Jor2jCut  :  120147 Events
```

@Anusreevijay769 you should also get the same.

ram1123 committed:
Updated Trigger list for 2l2q channel; Added mll cut of 40-180 GeV, and added option for output log file path in condor script

Anusreevijay769 committed:
Merge pull request #2 from ram1123/HZZ_Analysis_2l2q_v2_devR

Update Trigger list and mll cut

ram1123 committed:
Added argparser to GetLogSummary.py script; Added 2017 trigger list to Helper.py script; Added UL2017 root file path to post_proc.py

ram1123 committed:
Merge branch 'HZZ_Analysis_2l2q_v2_dev' of github.com:Anusreevijay769/nanoAOD_skim into HZZ_Analysis_2l2q_v2_devR

ram1123 committed:
Merge remote-tracking branch 'upstream/HZZ_Analysis' into HZZ_Analysis_2l2q_v2_devR

ram1123 committed:
- Updated 2018 trigger list for 202q

- Added option in post-proc such that it won't break the condor job

Anusreevijay769 committed:
made compatible with python2

Anusreevijay769 committed:
making compatible with python2

ram1123 committed:
Merge branch 'HZZ_Analysis_2l2q_v2_dev' of github.com:Anusreevijay769/nanoAOD_skim into HZZ_Analysis_2l2q_v2_devR

Anusreevijay769 committed:
Merge pull request #3 from ram1123/HZZ_Analysis_2l2q_v2_devR

Update for 2l2q by merging PR#2

Anusreevijay769 committed:
added new root file

Anusreevijay769 committed:
added new root file

ram1123 commented on post_proc.py:
```diff
@@ -7,34 +7,52 @@
+    testfile = "root://cms-xrd-global.cern.ch//store/mc/RunIISummer20UL18NanoAODv9/GluGluHToZZTo4L_M125_TuneCP5_13TeV_powheg2_JHUGenV7011_pythia8/NANOAODSIM/106X_upgrade2018_realistic_v16_L1v1-v1/120000/3ED05633-EBB7-4A44-8F9D-CD956490BCFD.root"
+    testfilelist.append(testfile)
+
+if WhichSample == "4l":
```
Here we need to add else conditon:

if ifRunningOnCondor is true then it should not enter this.

ram1123 commented:
Hi @YujiLee301 ,

Review this PR and drop your comments if you have any. Thanks.

Anusreevijay769 committed:
Updated utils files; made compitable with python2

Anusreevijay769 committed:
added energy, pt, eta and phi for 2j system

Anusreevijay769 committed:
updated 2j part to ZZ system

Anusreevijay769 committed:
updated 2j part to ZZ system

Anusreevijay769 committed:
updated file

Anusreevijay769 committed:
updated file

Anusreevijay769 committed:
changes for 2l2nu branch

Anusreevijay769 committed:
update to the 2l2nu part

Anusreevijay769 committed:
Merge pull request #14 from Anusreevijay769/HZZ_Analysis_2l2v_dev

Adding HZZ->2l2nu Category

ram1123 committed:
Revert "Adding HZZ->2l2nu Category" (#15)

ram1123 committed:
- Removed the submodule utils

ram1123 committed:
- Minor update in condor submission script.
- Added summary of additional script information in the README file

ram1123 committed:
Added golden json information to post-proc

ram1123 committed:
Merge branch 'HZZ_Analysis' of github.com:ram1123/nanoAOD_skim into HZZ_Analysis_2l2q_v2_devR

ram1123 committed:
Merge branch 'RemoveUtilsSubmodule' of github.com:ram1123/nanoAOD_skim into HZZ_Analysis_2l2q_v2_devR

ram1123 committed:
Merge branch 'HZZ_Analysis_2l2q_v2_devR' of github.com:ram1123/nanoAOD_skim into HZZ_Analysis_2l2q_v2_devR

ram1123 committed:
- Few files changed "nanoAOD_vvVBS" -> "nanoAOD_skim"
- Added golden josn file (last commit there was a bug)
- Added JME corrector
- Added PU SF module
- Added bTag SF module (For now its not working)

Anusreevijay769 committed:
Merge pull request #4 from ram1123/HZZ_Analysis_2l2q_v2_devR

Golden JSON, cuts to YAML

Anusreevijay769 committed:
added data root file

Anusreevijay769 committed:
update

ram1123 committed:
Hzz analysis 2l2q v2 dev r (#5)

* Add golden json & fix submodule utils (#16)

* - Removed the submodule utils

* - Minor update in condor submission script.
- Added summary of additional script information in the README file

* Added golden json information to post-proc

* - replaced string "nanoAOD_vvVBS"  to "nanoAOD_skim" in few files
- Fix the json bug (For MC we don't need to provide)
- Added modules createJMECorrector, btagSFProducer, etc

* - Still issue with the btagSF
- Updated the method for modules to run
- Added PUWeightProducer

* Added a text file having list of MC samples

* Added additional cuts for 2l2q channel

* - Improved post-proc for the condor jobs submission

* Fix a bug in root file name

Anusreevijay769 committed:
Fix merge conflict

Anusreevijay769 committed:
Fix merge conflict

ram1123 commented on H4LCppModule.py:
```diff
@@ -59,31 +63,54 @@ def beginJob(self):
+        print("{:27}:{:7} {}".format("Pass4eCut: ", str(self.worker.cut4e), " Events"))
         print("Pass4eGhostRemoval: "+str(self.worker.cutghost4e)+" Events")
         print("Pass4eLepPtCut: "+str(self.worker.cutLepPt4e)+" Events")
         print("Pass4eQCDSupress: "+str(self.worker.cutQCD4e)+" Events")
```
Update the above three lines similar pattern what we used for all others. I mean like below

```python
print("{:27}:{:7} {}".format("Pass4eCut: ", str(self.worker.cut4e), " Events"))
```

ram1123 commented on H4LCppModule.py:
```diff
@@ -59,31 +63,54 @@ def beginJob(self):
+        print("{:27}:{:7} {}".format("Pass4muCut: ", str(self.worker.cut4mu), " Events"))
         print("Pass4muGhostRemoval: "+str(self.worker.cutghost4mu)+" Events")
         print("Pass4muLepPtCut: "+str(self.worker.cutLepPt4mu)+" Events")
         print("Pass4muQCDSupress: "+str(self.worker.cutQCD4mu)+" Events")
```
Update the above three lines similar pattern what we used for all others. I mean like below

```python
print("{:27}:{:7} {}".format("Pass4eCut: ", str(self.worker.cut4e), " Events"))
```

ram1123 commented on input_data_Files/sample_list_v9_2018_campaign.dat:
Delete this file. We already have another file for this.

Anusreevijay769 committed:
input file updated to run over data and mc

Anusreevijay769 committed:
updates for running over data sample

Anusreevijay769 committed:
updates

Anusreevijay769 committed:
update for nElectron/nMuon issue

Anusreevijay769 committed:
update

Anusreevijay769 committed:
updates

Anusreevijay769 committed:
updates

Anusreevijay769 committed:
Fix merge conflict with master branch

Anusreevijay769 committed:
Debuging the cut-flow between h4l and h2l2q.. bug is still there

ram1123 committed:
Update H4LCppModule.py

Fixed the bug of bool branch passZZ*

Anusreevijay769 committed:
Hzz analysis 2l2v dev (#6)

* corrected TLorentz vector for MET part

* updates

* updates

* updates

* updated with MET part

* updated with MET part

* updates

* updates

* Temporary

* updates

* updates

* update for merging 2l2q and 2l2nu

* Fixed lepton selection

Anusreevijay769 committed:
Fix merge conflict

Anusreevijay769 committed:
updates

ram1123 committed:
BUGFIX: Leading two leptons kinematics were identical for all events (#7)

* Fixed the bug of bool branch passZZ*

* RoccoR with official nano-tools (#27)

* RoccoRUpdate

* Delete KalmanMuonCalibrationsProducer/data/roccor.Run2.v5 directory

---------

Co-authored-by: yuji <yuji.li@cern.ch>

* mergeOutput.py

* Some minor changes

* Added modules/functions for MET filters and GEN variables (#20)

* - Improved post-proc for the condor jobs submission

* Fix a bug in root file name

* Added the GEN producer and the MET filter

* - Added total entries
- Updated event summary print statement

* Committed changes without testing

* Updated keep_and_drop files

* Added METfilter in cut-flow

* Updated README

* Updating formating mainly to improve readebility

* Improved working of "--NOsyst" option

* - BUGFIX: Each leptons had same pT value. This is because we forgot the clear the vector.
- Added DEBUG option
- Updated condor job such that it can run over sl7 machine

* BugFix: We need to initialize TLorentzVector everytime by 0

---------

Co-authored-by: YujiLee301 <99654114+YujiLee301@users.noreply.github.com>
Co-authored-by: yuji <yuji.li@cern.ch>

ram1123 committed:
Update for 2l2nu and Bug fix for 2l2q channel (#9)

* RoccoR with official nano-tools (#27)

* RoccoRUpdate

* Delete KalmanMuonCalibrationsProducer/data/roccor.Run2.v5 directory

---------

Co-authored-by: yuji <yuji.li@cern.ch>

* # For 2l2nu channel

- Added b-tagged jet info
- Added VBF jets and VBF di-jet kinematics. Along with the VBF jet index

# For 2l2q channel

BUGFIX-1: Updated "jetidx.size()+1" with "jetidx.size()" to select di-jet

BUBFIX-2: Jet index was wrong while accessing jets info when selecting based on mass

BUGFIX-3: Also, fixed jet index that was saved in output branch

* Added deltaPhi var for 2l2nu channel

---------

Co-authored-by: YujiLee301 <99654114+YujiLee301@users.noreply.github.com>
Co-authored-by: yuji <yuji.li@cern.ch>

ram1123 committed:
RoccoR with official nano-tools (#27) (#10)

* RoccoRUpdate

* Delete KalmanMuonCalibrationsProducer/data/roccor.Run2.v5 directory

---------

Co-authored-by: YujiLee301 <99654114+YujiLee301@users.noreply.github.com>
Co-authored-by: yuji <yuji.li@cern.ch>

Anusreevijay769 committed:
Fix merge conflict with master branch HZZ_Analysis

ram1123 committed:
Some fixes (#11)

* Update condor job info and keep_and_drop file

* Fix: Size of output root file

Details here: https://github.com/ram1123/nanoAOD_skim/issues/38

* - Updated condor job script
- Updated condor job resubmit script
- Updated the keep and drop file- reduced the list
- Updated the merge nanoAOD root file script
- Updated README file

* - Updated condor setup script
- Updated condor resubmit script
- Added nanoAOD root file hadd-ed script

* Updated keep_and_drop*.txt

* - Removed the hardcoded output root file name in the post-proc and in the condor job script

* Added few scripts; later remove once that are not necessary or obselete

* Running version after update 2l2nu function

* Added some new variables to the output and naming convention

* Some updates to workflow of 2l2nu and 2l2q

* Removed the hardcoded cut-flow json file name and added option "--Nosyst" in condor script so that one can submit jobs without systematics

* - Updated 2018 YAML file to incorporate the separate trigger list for the different final states
- Accordingly update the code

Anusreevijay769 committed:
updates

Anusreevijay769 committed:
updates for merge conflict

Anusreevijay769 committed:
selection of z1 candidate for emu control region

Anusreevijay769 committed:
emu controlregion - lepton pt and eta update

Anusreevijay769 committed:
updates for emu control region

Anusreevijay769 committed:
updates

Anusreevijay769 committed:
update to find delta boost between two Z bosons

Anusreevijay769 committed:
Pz of neutrino

Anusreevijay769 committed:
update

Anusreevijay769 committed:
change of initial values of njets

ram1123 committed:
Merge branch and some minor changes (#12)

* Update condor job info and keep_and_drop file

* Fix: Size of output root file

Details here: https://github.com/ram1123/nanoAOD_skim/issues/38

* - Updated condor job script
- Updated condor job resubmit script
- Updated the keep and drop file- reduced the list
- Updated the merge nanoAOD root file script
- Updated README file

* - Updated condor setup script
- Updated condor resubmit script
- Added nanoAOD root file hadd-ed script

* Updated keep_and_drop*.txt

* - Removed the hardcoded output root file name in the post-proc and in the condor job script

* Added few scripts; later remove once that are not necessary or obselete

* Running version after update 2l2nu function

* Added some new variables to the output and naming convention

* Some updates to workflow of 2l2nu and 2l2q

* Removed the hardcoded cut-flow json file name and added option "--Nosyst" in condor script so that one can submit jobs without systematics

* - Updated 2018 YAML file to incorporate the separate trigger list for the different final states
- Accordingly update the code

* - BugFix: Mass windown of leptonic Z boson was missing the `fabs` function, which caused the window to be incorrectly defined
- BugFix: nJets branches should be initialize with ZERO, but it was initialize with -999
- Removed b-tag cut from 2l2nu selection
- Moved the hardcoded b-tag cut to the configuration file
- Added missing background samples

* - BugFix: Condor job not using custom name for the cut-flow file
- BugFix: If two sample names are same then it checks for word `ext` in the name to differentiate them

* Added script to remove duplicates from root file

* Updated condor for the output dir naming convention

* To avoid updating two keep and drop file I made two list for general variable and MC variables and keep them in .py file. Now, we need to write any variable only once in proper list GEN only or Data_MC.

Anusreevijay769 committed:
update lepton id

Anusreevijay769 committed:
update

Anusreevijay769 committed:
update lepton id

Anusreevijay769 committed:
update lepton id

Anusreevijay769 committed:
update lepton id

Anusreevijay769 committed:
updates for lepton id

Anusreevijay769 committed:
update on object selection

Anusreevijay769 committed:
update for jet_puid for pt < 50

Anusreevijay769 committed:
update for jet_puid for pt < 50

Anusreevijay769 committed:
update for jet_puid for pt < 50

Anusreevijay769 committed:
update for object selection

Anusreevijay769 committed:
update for gen part

Anusreevijay769 committed:
update for gen part

Anusreevijay769 committed:
update for gen part

Anusreevijay769 committed:
update to the gen part

Anusreevijay769 committed:
update to pz of neutrino

Anusreevijay769 committed:
list of signal dataset for different mass point

ram1123 committed:
Bug fix and condor improvement (#13)

* Removed transfer file from condor as now we don't have any files to transfer

* Now condor job will take arguments from txt file instead of many jdl files #51

* Update .write method to file instead of multi-string now just using one string using """

* As yaml-cpp is not available on el7 container so now cloe the yaml-cpp package and build it to remove the official dependencies. This solves Issue #54

* - Fix the ele eta cut for barrel and endcap transition
- Added the print statement under debug condition
- Added all high mass 2l2nu signal sample

* Changing the setup to work with EL9 #56

* Added variables to yml file for studying Di and single lepton triggers. (#57)

* Added variables to yml file for studying Di and single lepton triggers.

* Added leptons high pT triggers

---------

Co-authored-by: skeshri <sumit.keshri@cern.ch>

* Update condor script

* Fixed the library issue and nanohadd script fix for python2 to 3 shift

* Update condor job script

* Added output Dir option when running locally

* BugFix: Condor resubmit script is adapted as per the changes made for improving condor script #51 #56

---------

Co-authored-by: skeshri <sumitkesri@gmail.com>
Co-authored-by: skeshri <sumit.keshri@cern.ch>

ram1123 committed:
Updated README (#14)

ram1123 committed:
- Bugfix for GEN variables module. It was also triggering for the data but it should not. (#15)

- Added VBF MC sample in the sample list
- Updated the README
